parachains/runtimes/collectives/collectives-polkadot/src/ambassador/origins.rs

```diff
@@ -0,0 +1,125 @@
+// Copyright (C) 2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! The Ambassador Program's origins.
+
+#[frame_support::pallet]
+pub mod pallet_origins {
+	use crate::ambassador::ranks;
+	use frame_support::pallet_prelude::*;
+	use pallet_ranked_collective::Rank;
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(PhantomData<T>);
+
+	/// The pallet configuration trait.
+	#[pallet::config]
+	pub trait Config: frame_system::Config {}
+
+	#[derive(PartialEq, Eq, Clone, MaxEncodedLen, Encode, Decode, TypeInfo, RuntimeDebug)]
+	#[pallet::origin]
+	pub enum Origin {
+		/// Plurality voice of the [ranks::AMBASSADOR_TIER_1] members or above given via referendum.
+		AmbassadorTier1,
+		/// Plurality voice of the [ranks::AMBASSADOR_TIER_2] members or above given via referendum.
+		AmbassadorTier2,
+		/// Plurality voice of the [ranks::SENIOR_AMBASSADOR_TIER_3] members or above given via referendum.
+		SeniorAmbassadorTier3,
+		/// Plurality voice of the [ranks::SENIOR_AMBASSADOR_TIER_4] members or above given via referendum.
+		SeniorAmbassadorTier4,
+		/// Plurality voice of the [ranks::HEAD_AMBASSADOR_TIER_5] members or above given via referendum.
+		HeadAmbassadorTier5,
+		/// Plurality voice of the [ranks::HEAD_AMBASSADOR_TIER_6] members or above given via referendum.
+		HeadAmbassadorTier6,
+		/// Plurality voice of the [ranks::HEAD_AMBASSADOR_TIER_7] members or above given via referendum.
+		HeadAmbassadorTier7,
+		/// Plurality voice of the [ranks::MASTER_AMBASSADOR_TIER_8] members or above given via referendum.
+		MasterAmbassadorTier8,
+		/// Plurality voice of the [ranks::MASTER_AMBASSADOR_TIER_9] members or above given via referendum.
+		MasterAmbassadorTier9,
```

Comment:
I like how the Fellowship does this:
```suggestion
		/// Plurality voice of the [ranks::AMBASSADOR_TIER_1] members or above given via referendum.
		Ambassadors,
		/// Plurality voice of the [ranks::AMBASSADOR_TIER_2] members or above given via referendum.
		AmbassadorTier2,
		/// Plurality voice of the [ranks::SENIOR_AMBASSADOR_TIER_3] members or above given via referendum.
		SeniorAmbassadors,
		/// Plurality voice of the [ranks::SENIOR_AMBASSADOR_TIER_4] members or above given via referendum.
		SeniorAmbassadorTier4,
		/// Plurality voice of the [ranks::HEAD_AMBASSADOR_TIER_5] members or above given via referendum.
		HeadAmbassadors,
		/// Plurality voice of the [ranks::HEAD_AMBASSADOR_TIER_6] members or above given via referendum.
		HeadAmbassadorTier6,
		/// Plurality voice of the [ranks::HEAD_AMBASSADOR_TIER_7] members or above given via referendum.
		HeadAmbassadorTier7,
		/// Plurality voice of the [ranks::MASTER_AMBASSADOR_TIER_8] members or above given via referendum.
		MasterAmbassadors,
		/// Plurality voice of the [ranks::MASTER_AMBASSADOR_TIER_9] members or above given via referendum.
		MasterAmbassadorTier9,
```